src/com/redhat/ceylon/model/typechecker/model/Package.java

```diff
@@ -221,6 +222,8 @@ public TypeDeclaration getInheritingDeclaration(Declaration d) {
                 getModule()
                     .getAvailableDeclarations(
                             startingWith, proximity);
+        Map<String, DeclarationWithProximity> thingsToAdd = new HashMap<>();
```

Comment:
@bjansen y'know, you could just build a `List` of the `namesToAdd`. Not sure if that would be faster or slower.